### PR TITLE
Improved events validation

### LINKED ISSUE #148: Improve _events validation
- Raise a warning if the last (latest) event has onset before the halfway mark of the duration of the corresponding scan (number of volumes \* TR). The design is suspiciously short.
- Raise a warning if the last (latest) event has onset after the end of the duration of the corresponding scan. The design is suspiciously long.

### PR DESCRIPTION
**Pull request by olgn**
fixes #148 

* _events.tsv files should have the onset of the last event between values .5 * TR * number of volumes < onset < TR * number of volumes
* implements unit tests for this rule

### CODE REVIEW

File: tests/events.spec.js [hunk 1]
```diff
@@ -23,7 +25,58 @@ describe('Events', function(){
             events: ['/stimuli/images/red-square.jpg'],
             directory: [{relativePath: '/stimuli/images/red-square.jpg'}],
         };
-        validate.Events.checkStimuli(stimuli, issues);
+        validate.Events.Events([], stimuli, [], {}, issues);
         assert(issues.length === 0);
     });
+
+    it('should throw an issue if the onset of the last event in _events.tsv is more than TR * number of volumes in corresponding nifti header', function () {
+        let issues = [];
```
Comment (nellh) [inline tests/events.spec.js]: This can also be const, since you don't reassign it.

File: validators/events.js [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
 
-var checkStimuli = function (stimuli, issues) {
+const Events = function(events, stimuli, headers, jsonContents, issues) {
+    // check that all stimuli files present in /stimuli are included in an _events.tsv file
+    checkStimuli(stimuli, issues);
+
+    // check the events file for suspiciously long or short durations
+    checkDesignLength(events, headers, jsonContents, issues);
+
+    return;
```
Comment (nellh) [inline validators/events.js]: This return doesn't do anything.

File: validators/events.js [hunk 3]
```diff
@@ -19,6 +31,69 @@ var checkStimuli = function (stimuli, issues) {
...
+            if (lastEventOnset < shortDurationThreshold) {
+                issues.push(new Issue({
+                    file: event.file,
+                    code: 86
+                }));
+            }
+
+   
+        }
+    }, () => {return;});
```
Comment (nellh) [inline validators/events.js]: It's clearer to omit the callback here since it does nothing.

File: validators/events.js [hunk 4]
```diff
@@ -1,7 +1,19 @@
 /* eslint-disable no-unused-vars */
-var Issue = require('../utils').issues.Issue;
+const async = require('async');
+const utils = require('../utils');
+const Issue = utils.issues.Issue;
 
-var checkStimuli = function (stimuli, issues) {
+const Events = function(events, stimuli, headers, jsonContents, issues) {
```
Comment (nellh) [inline validators/events.js]: This isn't a constructor, it should be named events() or something more descriptive.

**Review by nellh — CHANGES_REQUESTED** [review]

**Comment by olgn** [comment]
@nellh all better :package:

**Review by nellh — APPROVED** [review]

**Pull request merged**